pandas/io/pytables.py

```diff
@@ -313,10 +313,14 @@ def read_hdf(path_or_buf, key=None, **kwargs):
 
         """
 
-    if kwargs.get('mode', 'a') not in ['r', 'r+', 'a']:
+    # Ignore any supplied mode, we only need to read data
```

Comment:
Please explain to me why the read_hdf function would ever want to use any mode other than 'r'. It's not like the returned object is linked to the file by an mmap - the file is opened, data are read, the file is closed. At no point is there any need to write to the file.